src/main/java/io/confluent/copycat/hdfs/HdfsSinkTask.java

```diff
@@ -43,37 +43,51 @@ public HdfsSinkTask() {
   public void start(Properties props) {
     try {
       HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
-      hdfsWriter = new HdfsWriter(connectorConfig);
-      context.resetOffset(hdfsWriter.getPreviousOffsets());
+      assignment = context.assignment();
+      hdfsWriter = new HdfsWriter(connectorConfig, assignment);
+      recover(assignment);
+      resetOffsets();
     } catch (ConfigException e) {
-      throw new CopycatRuntimeException(
+      throw new CopycatException(
           "Couldn't start HdfsSinkConnector due to configuration error.", e);
     }
+
   }
 
   @Override
   public void stop() throws CopycatException {
-    try {
-      hdfsWriter.close();
-    } catch (IOException e) {
-      throw new CopycatRuntimeException("Error closing the HDFS writer", e);
-    }
+    hdfsWriter.close();
+
   }
 
   @Override
   public void put(Collection<SinkRecord> records) throws CopycatException {
     String topic;
     int partition;
-    for (SinkRecord record: records) {
-      topic = record.getTopic();
-      partition = record.getPartition();
+    for (SinkRecord record : records) {
+      topic = record.topic();
+      partition = record.kafkaPartition();
       TopicPartition topicPart = new TopicPartition(topic, partition);
-      hdfsWriter.writeRecord(topicPart, record);
+      if (assignment.contains(topicPart)) {
+        hdfsWriter.writeRecord(topicPart, record);
+      } else {
+        log.warn("Seen record {} from not assigned TopicPartition {}", record, topicPart);
```

Comment:
Seen -> Saw. Should we also log this at a higher level, like `error`? This is a pretty critical error. I'm not even sure we should proceed if we hit this because it means we could be losing data since this method just ignores the data and moves on. It either indicates a serious problem with the underlying consumer or that something got messed up in this `HdfsSinkTask` code.